src/main/java/org/openrewrite/kotlin/internal/KotlinParserVisitor.java

```diff
@@ -3670,21 +3672,23 @@ public J visitElement(@Nullable FirElement firElement, ExecutionContext ctx) {
 
         int saveCursor = cursor;
         Space prefix = whitespace();
-        ASTNode node = nodes.get(cursor);
-        if (node != null) {
-            switch (node.getElementType().getDebugName()) {
+
+        List<PsiTree.Node> nodes = psiTree.findByCursor(cursor);
```

Comment:
This appears to visit the entire PSI per FIR element to collect a list of PSI nodes, then iterates through the entire list again?